operators/src/plot/temporal_vector_line_plot.rs

```diff
@@ -0,0 +1,360 @@
+        Operator, PlotOperator, PlotQueryProcessor, PlotResultDescriptor, QueryContext,
+        QueryRectangle, TypedPlotQueryProcessor, VectorQueryProcessor,
+    },
+    error::Error,
+};
+use async_trait::async_trait;
+use futures::StreamExt;
+use geoengine_datatypes::primitives::{FeatureDataType, FeatureDataValue};
+use geoengine_datatypes::{
+    collections::FeatureCollection,
+    plots::{Plot, PlotData},
+};
+use geoengine_datatypes::{
+    collections::FeatureCollectionInfos,
+    plots::{DataPoint, MultiLineChart},
+};
+use geoengine_datatypes::{
+    primitives::{Geometry, Measurement, TimeInterval},
+    util::arrow::ArrowTyped,
+};
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, ResultExt};
+use std::collections::HashMap;
+use std::{
+    cmp::Ordering,
+    collections::hash_map::Entry::{Occupied, Vacant},
+};
+
+pub const FEATURE_ATTRIBUTE_OVER_TIME_NAME: &str = "Feature Attribute over Time";
+const MAX_FEATURES: usize = 20;
+
+/// A plot that shows the value of an feature attribute over time.
+pub type FeatureAttributeValuesOverTime = Operator<FeatureAttributeValuesOverTimeParams>;
+
+/// The parameter spec for `FeatureAttributeValuesOverTime`
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FeatureAttributeValuesOverTimeParams {
+    pub id_column: String,
+    pub value_column: String,
+}
+
+#[typetag::serde]
+impl PlotOperator for FeatureAttributeValuesOverTime {
+    fn initialize(
+        mut self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<InitializedPlotOperator>> {
+        ensure!(
+            self.vector_sources.len() == 1,
+            error::InvalidNumberOfVectorInputs {
+                expected: 1..2,
+                found: self.vector_sources.len()
+            }
+        );
+        ensure!(
+            self.raster_sources.is_empty(),
+            error::InvalidNumberOfVectorInputs {
+                expected: 0..1,
+                found: self.raster_sources.len()
+            }
+        );
+
+        let source = self.vector_sources.remove(0).initialize(context)?;
+        let result_descriptor = source.result_descriptor();
+        let columns: &HashMap<String, FeatureDataType> = &result_descriptor.columns;
+
+        ensure!(
+            columns.contains_key(&self.params.id_column),
+            error::ColumnDoesNotExist {
+                column: self.params.id_column.clone()
+            }
+        );
+
+        ensure!(
+            columns.contains_key(&self.params.value_column),
+            error::ColumnDoesNotExist {
+                column: self.params.value_column.clone()
+            }
+        );
+
+        let id_type = columns.get(&self.params.id_column).expect("checked");
+        let value_type = columns.get(&self.params.value_column).expect("checked");
+
+        ensure!(
+            id_type == &FeatureDataType::Text || id_type == &FeatureDataType::Int,
```

Comment:
I guess categorical would be optimal, here, but I guess a TODO sufficies.

Reply:
I don't think Category would be optimal. We really would need a way of flagging an attribute as an id (at source level, and operators would need to ensure that it stays an id or remove the flag). A categorical value does not necessarily uniquely identify a feature over time. Rather than a todo we would need to add a card.

Reply:
Hm, that's true, but we could offer an aggregate over a category.

Reply:
626feaf41576cd7e04f859ea9701a2714356ad71